_pages/doc/contributing-to-llvm.md

```diff
@@ -0,0 +1,168 @@
+---
+title: "Contributing to LLVM"
+---
+
+## Context
+
+Intended audience:
+
+- Those who want to contribute improvements or fixes to the LLVM M68k backend
+- Experienced computer users
+- Experienced C++ programmers
+- Those familiar with git, GitHub and the command line
+
+## Setup
+
+Perequisites:
+
+- A Unix-like OS
+- A git client
+- A native C++ compiler
+- CMake
+- Ninja build system
+- A C++ code editor
+
+### Forking and cloning the LLVM repository
+
+Head over to [the LLVM repository page](https://github.com/llvm/llvm-project) and fork it.
+
+Once GitHub has finished forking the repo, use git to clone your fork.
+
+```sh
+git clone git@github.com:<your-github-username>/llvm-project.git
+cd llvm-project
+mkdir build
+```
+
+### Generating build files using CMake
+
+Invoke CMake to generate the build files:
+
+```sh
+cmake \
+    -B build \
+    -D BUILD_SHARED_LIBS="On" \
+    -D CMAKE_BUILD_TYPE="Release" \
+    -D LLVM_ENABLE_ASSERTIONS="On" \
+    -D LLVM_ENABLE_PROJECTS="" \
+    -D LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="M68k" \
+    -D LLVM_TARGETS_TO_BUILD="" \
+    -D LLVM_USE_SPLIT_DWARF="On" \
+    -G Ninja \
+    -S llvm
+```
+
+#### Rationale for the CMake options
+
+CMake usage can be displayed with `cmake --help`.
+
+The LLVM CMake options which affect the build are [documented here](https://llvm.org/docs/CMake.html).
+
+`-B build`
+
+We'll use a directory named `build` to store build files. This directory is already ignored by the `.gitignore` file, so it's assumed to be the standard build directory.
+
+`-D BUILD_SHARED_LIBS="On"`
+
+We'll build shared libraries rather than static libraries, to save some disk space.
+
+`-D CMAKE_BUILD_TYPE="Release"`
+
+We'll build the `Release` configuration, as otherwise running tests can be quite slow.
+
+**TODO:** Should we build the `RelWithDebInfo` configuration instead?
+
+`-D LLVM_ENABLE_ASSERTIONS="On"`
+
+We'll enable assertions, in order to catch more errors during runtime.
+
+`-D LLVM_ENABLE_PROJECTS=""`
+
+We'll remove unnecessary projects from the build.
```

Comment:
I think people usually wants to build clang as well. Maybe we can include clang here?